client/src/stores/parflow.js

```diff
@@ -90,48 +84,30 @@ export default {
       state.busyCount += 1;
 
       const job = getters.PARFLOW_JOB;
-      const { application, runId } = job;
 
       if (job.runReset) {
         await dispatch('PVW_RESET');
       }
 
       // Push configuration
       await dispatch('PVW_CONFIG_UPDATE', job);
+    },
+    PARFLOW_RUN_COMPLETE({ state, getters, dispatch }, { returncode }) {
+      state.busyCount -= 1;
 
-      const task = JSON.stringify({ application, runId });
-      return new Promise((resolve, reject) => {
-        const xhr = new XMLHttpRequest();
-
-        xhr.onreadystatechange = (e) => {
-          if (xhr.readyState === 4) {
-            state.busyCount -= 1;
-            if (job.runReset) {
-              state.execCount = 1;
-            } else {
-              state.execCount += 1;
-            }
-            if (xhr.status === 200 || xhr.status === 0) {
-              state.runReset = 0;
-              dispatch('PARFLOW_RESET_WELLS');
-              resolve(JSON.parse(xhr.responseText));
-            } else {
-              reject({ xhr, e });
-            }
-          }
-        };
-
-        // Make request
-        if (userParams.dev) {
-          xhr.open('POST', 'http://localhost:9000/paraview/', true);
-        } else {
-          xhr.open('POST', window.SANDTANK_SESSION_MANAGER_URL, true);
-        }
+      const job = getters.PARFLOW_JOB;
+      if (job.runReset) {
+        state.execCount = 1;
+      } else {
+        state.execCount += 1;
+      }
 
-        xhr.setRequestHeader('Content-Type', 'text/plain;charset=UTF-8');
-        xhr.responseType = 'text';
-        xhr.send(task);
-      });
+      if (returncode === 0) {
```

Comment:
So, I think the [`runSimulation.sh`](https://github.com/hydroframe/SandTank/blob/master/deploy/pvw/simulations/runSimulation.sh) script is always returning zero right now, because [the last line is a sleep command](https://github.com/hydroframe/SandTank/blob/70ab6128049f70994c446bc3f88435bc54106ae7/deploy/pvw/simulations/runSimulation.sh#L45). We might want to `set -e` at the top of the script if we want this returncode checking to work.

But if it's not a big issue, we don't need to.

Reply:
you can remove the sleep call now that you are doing it in python. 
Adding `set -e` make sense as well.

Reply:
Done.